numba_cuda/numba/cuda/simulator/__init__.py

```diff
@@ -38,11 +38,20 @@
     sys.modules["numba.cuda.cudadrv.devicearray"] = cudadrv.devicearray
     sys.modules["numba.cuda.cudadrv.devices"] = cudadrv.devices
     sys.modules["numba.cuda.cudadrv.driver"] = cudadrv.driver
+    sys.modules["numba.cuda.cudadrv.linkable_code"] = cudadrv.linkable_code
     sys.modules["numba.cuda.cudadrv.runtime"] = cudadrv.runtime
     sys.modules["numba.cuda.cudadrv.drvapi"] = cudadrv.drvapi
     sys.modules["numba.cuda.cudadrv.error"] = cudadrv.error
     sys.modules["numba.cuda.cudadrv.nvvm"] = cudadrv.nvvm
 
-    from . import compiler
+    from . import bf16, compiler, _internal
 
+    sys.modules["numba.cuda.bf16"] = bf16
     sys.modules["numba.cuda.compiler"] = compiler
+    sys.modules["numba.cuda._internal"] = _internal
+    sys.modules["numba.cuda._internal.cuda_bf16"] = _internal.cuda_bf16
+
+    from numba.cuda.simulator import memory_management
+
+    sys.modules["numba.cuda.memory_management"] = memory_management
+    sys.modules["numba.cuda.memory_management.nrt"] = memory_management.nrt
```

Comment:
there's something about this that's making me scratch my head. Here we're adding modules in a way that seems like it's making them available for the CUDA simulator, but later on it looks like the CUDA simulator disables NRT globally. Are these just here so everything imports correctly, even if they're not used?

Reply:
They're here so that users of these modules / packages can import them without error (e.g. tests, user code, etc.). It's important that anything that might get imported has a dummy entry in `sys.modules` so that the real module never gets imported later (e.g. if we didn't populate `sys.modules` with dummy ones) because then we'd end up with bits of simulator imported and bits of the real target, and that very quickly leads to a mess that crashes with a surprising exception.